### PR TITLE
Docs: add Windows binaries available on GISInternals

### PR DESCRIPTION
**Pull request by mloskot**
## What does this PR do?

Port link from https://trac.osgeo.org/gdal/wiki/DownloadingGdalBinaries to Windows binaries offered by @szekerest 

## Tasklist

 - [ ] All CI builds and checks have passed

### CODE REVIEW

File: gdal/doc/source/download.rst [hunk 1]
```diff
@@ -59,8 +59,12 @@ Windows builds are available via `Conda Forge`_ (64-bit only). See the
 
 .. _`Conda Forge`: https://anaconda.org/conda-forge/gdal
 
+Alternatively, [Tamas Szekeres](https://github.com/szekerest) provides packages
+with daily builds of latest stable releases as well as maintenance and
+development branches on his `GISInternals`_ site.
+
+.. _`GISInternals`: https://www.gisinternals.com
```
Comment (hobu) [inline gdal/doc/source/download.rst]: If we're adding this, we should add MS4W and any others that want to be added too. See #1641 for discussion.

Comment (mloskot) [reply gdal/doc/source/download.rst]: I actually wondered why none of those are listed in the new docs and I decided to provoke some answers with this PR (I failed to find the other one for MS4W). 

In #1641 you say:
> I think we only want to list builds that have fully reproducible open source build recipes

It makes sense. I'm closing this PR and leaving it to @szekerest to re-open, if he feels determined to have his packages advertised.

**Comment by mloskot** [comment]
Dropping as per https://github.com/OSGeo/gdal/pull/1866#discussion_r326299551

**Comment by rouault** [comment]
Could make sense to clarify the scope of the section and add "which have fully reproducible open source build recipes" after "In this section we list a number of the binary distributions of GDAL.". Either as user visible text, or as a ReST comment, for future contributors.

**Comment by mloskot** [comment]
@rouault #1868

**Pull request closed without merging**